File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/utils/RoomItemDiffCallback.java
package vng.zalo.tdtai.zalo.zalo.utils;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import vng.zalo.tdtai.zalo.zalo.models.RoomItem;

public class RoomItemDiffCallback extends DiffUtil.ItemCallback<RoomItem> {
    private static final String TAG = RoomItemDiffCallback.class.getSimpleName();

//    @Inject
    public RoomItemDiffCallback(){}

    @Override
    public boolean areItemsTheSame(@NonNull RoomItem oldItem, @NonNull RoomItem newItem) {
        return oldItem.roomId.equals(newItem.roomId);
    }

    @Override
    public boolean areContentsTheSame(@NonNull RoomItem oldItem, @NonNull RoomItem newItem) {
        return oldItem.equals(newItem);
    }
}

File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/utils/RoomItemDiffCallback.kt
package vng.zalo.tdtai.zalo.zalo.utils

import android.util.Log
import androidx.recyclerview.widget.DiffUtil
import vng.zalo.tdtai.zalo.zalo.models.RoomItem

class RoomItemDiffCallback : DiffUtil.ItemCallback<RoomItem>() {

    override fun areItemsTheSame(oldItem: RoomItem, newItem: RoomItem): Boolean {
        Log.d(TAG, if (oldItem.roomId == newItem.roomId) "same item" else "diff item")
        return oldItem.roomId == newItem.roomId
    }

    override fun areContentsTheSame(oldItem: RoomItem, newItem: RoomItem): Boolean {
        Log.d(TAG, if (oldItem == newItem) "same content" else "diff content")
        return oldItem == newItem
    }

    companion object{
        private val TAG = RoomItemDiffCallback::class.java.simpleName
    }
}